### PR TITLE
Sub headings+

### PR DESCRIPTION
**Pull request by MikeDodaro**
Add subheadings to the following articles:
Dirac notation
Pauli measurements
Circuits
Oracles

### CODE REVIEW

File: articles/techniques/operations-and-functions.md [hunk 1]
```diff
@@ -263,7 +263,7 @@ This new function is indeed a function, in that if we call it with the same valu
 Thus, the decoder can be safely run inside operations without having to reason about how the decoding logic interacts with the definitions of the different operation specializations.
 That is, we have isolated the classical logic inside a function, guaranteeing to the compiler that the function call can be reordered with impunity so long as the input is preserved.
 
-We can also treat functions as first-class values, as we will see in more detail when we discuss [operations and function types](#operation-and-function-types).
+We can also treat functions as first-class values, as we will see in more detail when we discuss [operations and function types](#operations-and-functions-as-first-class-values).
```
[the remark this answers is not in the document]
Comment (MikeDodaro) [reply articles/techniques/operations-and-functions.md]: Thanks @cgranade   GitHub protocol still throws me at times.  I think I now understand why #415 was pulled in here.

File: articles/concepts/dirac-notation.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ Of course evaluating this tensor product is impractical because the vector lies
 
 [*Dirac notation*](https://en.wikipedia.org/wiki/Bra%E2%80%93ket_notation) solves these issues by presenting a new language to fit the precise needs of quantum mechanics.  For this reason, we recommend the reader not view the examples in this section as a rigid prescription of how to describe quantum states, but rather encourage the reader to view these as suggestions that can be used to concisely express quantum ideas.
 
+## bra ket notation
```
[the remark this answers is not in the document]
Comment (MikeDodaro) [reply articles/concepts/dirac-notation.md]: This heading is probably not needed here, so I took it out.

File: articles/concepts/dirac-notation.md [hunk 3]
```diff
@@ -142,7 +149,9 @@ $$
 
 which matches the multi-qubit measurement discussion.  The generalization of this result to the multi-qubit case, however, is slightly more straightforward to express using Dirac notation than column-vector notation, and is entirely equivalent to the previous treatment.
 
+## State operator
```
[the remark this answers is not in the document]
Comment (MikeDodaro) [reply articles/concepts/dirac-notation.md]: Information in this paragraph refers to density operators, as that's a bit more of a common name than "state operators."

File: articles/concepts/dirac-notation.md [hunk 4]
```diff
@@ -25,42 +25,46 @@ More generally, if $\psi$ and $\phi$ are quantum state vectors their inner produ
...
 
-The following notation is often used to describe the states that result from applying the Hadamard gate to $|0\rangle$ and $|1\rangle$ (which correspond to the unit vectors in the $+x$ and $-x$ directions on the Bloch sphere):
+### Example: representing the Hadamard operation with Dirac notation
+
+The following notation is often used to describe the states that result from applying the Hadamard gate to $\ket{0}$ and $\ket{1}$ (which correspond to the unit vectors in the $+x$ and $-x$ directions on the Bloch sphere):
 
 $$
-\frac{1}{\sqrt{2}}\begin{bmatrix} 1 \\\\  1 \end{bmatrix}=H|0\rangle = |{+}\rangle,\qquad
-\frac{1}{\sqrt{2}}\begin{bmatrix} 1 \\\\  -1 \end{bmatrix} =H|1\rangle = |{-}\rangle .
+\frac{1}{\sqrt{2}}\begin{bmatrix} 1 \\\\  1 \end{bmatrix}=H\ket{0} = |{+}\rangle,\qquad
```
Comment (tcNickolas) [inline articles/concepts/dirac-notation.md]: There are still a lot of `|...\rangle` left in the formulas

Comment (cgranade) [reply articles/concepts/dirac-notation.md]: I'm not surprised, there's a lot of different variations that make it hard to find and replace. Thanks for catching this one!

**Pull request merged**